base/multidimensional.jl

```diff
@@ -863,3 +863,193 @@ If `dim` is specified, returns unique regions of the array `itr` along `dim`.
         @nref $N A d->d == dim ? sort!(uniquerows) : (1:size(A, d))
     end
 end
+
+"""
+    C, ia, ib, ic = uniqueind(itr, dim)
+
+A function that operates similiarly to `unique(itr,dim)` but returns multiple
+output arguments having the following properties:
+
+C - the unique elements of the array `itr` along the selected dimension `dim`
+ia - a Vector{Int} of indices such that:
```

Comment:
Choice of names `ia` and `ic` were based on the MATLAB(R) names but can certainly be changed as desired by the community.  Choice of `itr` was just based on making minimal changes from the existing `unique` function, but `A` would be a closer choice to the existing MATLAB name.

Along with `C`, the vector of vectors `ib` is essentially part of what is returned from the [`group`](http://code.kx.com/wiki/Reference/group) function in q, also known as the left (=) operator in K.  Returning this output is functionality that was specifically requested of me to implement, but is also why I mentioned above that it might be desired to have multiple functions returning different outputs instead of just one returning the four outputs included here.